lib/matplotlib/figure.py

```diff
@@ -2084,7 +2062,9 @@ def _do_layout(gs, mosaic, unique_ids, nested):
                     nested_output = _do_layout(
                         gs[j, k].subgridspec(rows, cols),
                         nested_mosaic,
-                        *_identify_keys_and_nested(nested_mosaic)
+                        *_identify_keys_and_nested(nested_mosaic),
+                        sub_add_func,
```

Comment:
If you did this as 

```
subadd_func if subfigs_for_nested.get(key) is None else subfigs_for_nested[key].add_subfigure:
```

then you would still only need to pass one piece of information into the recursive call

Reply:
Ok, I think I see what you mean. I changed it to what you suggested, plus the internal code for the `_get_subfigs_from_nested` and it seems to work. I'll test it a bit more and amend the PR if it looks good. Thank you!